test/browser/render.js

```diff
@@ -185,6 +185,27 @@ describe('render()', () => {
 			.and.matches(/position\s*:\s*relative\s*/);
 	});
 
+	it('string style should call setAttribute', () => {
+		const style = 'background-color: red;';
+		const spy = sinon.spy(Element.prototype, 'setAttribute');
+		render(<div style={style} />, scratch);
+		
+		expect(spy.firstCall.args).to
+			.deep.equal(['style', style]);
+		expect(scratch.childNodes[0]).to.have.deep.property('style.cssText')
+			.that.equals(style);
+		spy.restore();
+	});
+
+	it('object style prop should not call setAttribute', () => {
+		const style = { backgroundColor: 'papayawhip' };
```

Comment:
If it changes style attribute, as in your implementation, it should call the `setAttribute` right?

Reply:
no, `object` style props are passed to the object handler:

```
		if (value && typeof value==='object') {
			if (!isString(old)) {
				for (let i in old) if (!(i in value)) node.style[i] = '';
			}
			for (let i in value) {
				node.style[i] = typeof value[i]==='number' && !NON_DIMENSION_PROPS[i] ? (value[i]+'px') : value[i];
			}
		}
```